### PR TITLE
feat(query-service): Add cold storage support 

### PR DESCRIPTION
**Pull request by ahsanbarkati**
Add support for cold-storage in Signoz. Cold storage support is provided by clickhouse by its tiered storage system. We can have multi-tiered storage and can make clickhouse move data between these tiers based on TTL. This change creates the required APIs in signoz in order to setup the tiered storage and set required TTL.

Add the following API
- `/api/v1/disks`: A GET API to list down the disks visible to clickhouse.

Modify the following API:
- `/api/v1/settings/ttl`: Modifies the POST handler to take `coldStorage` (the storage to which we want the data to move after TTL) and `toColdDuration` (The duration after which we want the data to move to the cold storage)

### CODE REVIEW

File: pkg/query-service/app/clickhouseReader/reader.go [hunk 1]
```diff
@@ -2583,39 +2583,73 @@ func (r *ClickHouseReader) GetFilteredSpansAggregates(ctx context.Context, query
...
+			fmt.Errorf("error while setting ttl. ttl type should be <metrics|traces>, got %v",
+				params.Type)}
+	}
+
+	// Set the storage policy for the required table. If it is already set, then setting it again
+	// will not a problem.
+	if len(params.ColdStorageVolume) > 0 {
+		policyReq := fmt.Sprintf("ALTER TABLE %s MODIFY SETTING storage_policy='tiered'", tableName)
+
+		zap.S().Info("Executing Storage policy request: %s\n", policyReq)
```
Comment (prashant-shahi) [inline pkg/query-service/app/clickhouseReader/reader.go:2623]: > zap.S().Info(...

it should be `zap.S().Infof(...`.
Also, for I think this log should be a debug level.

**Review by prashant-shahi — APPROVED** [review]
:shipit: LGTM

File: pkg/query-service/go.mod [hunk 2]
```diff
@@ -4,7 +4,7 @@ go 1.14
 
 require (
 	cloud.google.com/go v0.88.0 // indirect
-	github.com/ClickHouse/clickhouse-go v1.4.5
+	github.com/ClickHouse/clickhouse-go v1.5.4
```
[the remark this answers is not in the document]
Comment (ankitnayan) [reply pkg/query-service/go.mod:7]: @ahsanbarkati we will be migrating to use `clickhouse-go/v2` in coming week. I don't think there will be a major change due to that.

**Review by ankitnayan — APPROVED** [review]

File: pkg/query-service/go.mod (hunk 2 above)
[the remark this answers is not in the document]
Comment (makeavish) [reply pkg/query-service/go.mod:7]: Yeah, shouldn't be an issue

**Review by makeavish — APPROVED** [review]

**Pull request merged**